pandas/core/generic.py

```diff
@@ -2058,6 +2058,25 @@ def __delitem__(self, key):
         except KeyError:
             pass
 
+    def _take(self, indices, axis=0, convert=True, is_copy=False):
+        """
+        The internal version of `self.take` that will house the `convert`
+        parameter for group-by purposes.
+        """
+
+        self._consolidate_inplace()
+        new_data = self._data.take(indices,
+                                   axis=self._get_block_manager_axis(axis),
+                                   verify=True)
+        result = self._constructor(new_data).__finalize__(self)
+
+        # Maybe set copy if we didn't actually change the index.
+        if is_copy:
+            if not result._get_axis(axis).equals(self._get_axis(axis)):
+                result._set_is_copy(self)
+
+        return result
+
     def take(self, indices, axis=0, convert=True, is_copy=True, **kwargs):
```

Comment:
also deprecate ``is_copy``; that is another internal parameter; its only necessary for ``_take``

Reply:
That's a fair point.  If you don't mind, I'll do this in a subsequent PR once the infrastructure for `_take` has been merged in.